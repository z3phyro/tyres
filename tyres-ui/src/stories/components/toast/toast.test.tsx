import { expect, test } from "vitest";
import { render } from "@solidjs/testing-library";
import Toast from "./toast";
import { Toast as KToast } from "@kobalte/core";
import { BorderColor, EUiVariant } from "~/core/types/ui-variants.type";

test("Toast renders", () => {
  const { getByRole } = render(() => <KToast.Region>
    <Toast title="Test title" toastId={1} />
  </KToast.Region>);

  const toastElement = getByRole("status");
  expect(toastElement).toBeTruthy();
});

test("Toast renders with description", () => {
  const { getByText } = render(() => <KToast.Region>
    <Toast title="Test title" description="Test description" toastId={1} />
  </KToast.Region>);

  const descriptionElement = getByText("Test description");
  expect(descriptionElement).toBeTruthy();
});

test("Toast renders with custom variant", () => {
  const { getByRole } = render(() => <KToast.Region>
    <Toast title="Test title" toastId={1} variant={EUiVariant.Success} />
  </KToast.Region>);

  const toastElement = getByRole("status");
  expect(toastElement).toHaveClass(BorderColor[EUiVariant.Success]);
});

test("Toast snapshot test", () => {
  const { asFragment } = render(() => <KToast.Region>
    <Toast title="Test title" description="Test description" toastId={1} variant={EUiVariant.Success} />
  </KToast.Region>);

  expect(asFragment()).toMatchSnapshot();
});
