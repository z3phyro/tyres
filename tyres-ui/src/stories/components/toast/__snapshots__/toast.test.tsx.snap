// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Toast snapshot test 1`] = `"<div role="region" tabindex="-1" aria-label="Notifications (alt+T)" style="pointer-events: none;" data-kb-top-layer="" id="toast-region-cl-6"><li role="status" tabindex="0" style="user-select: none;" aria-live="assertive" aria-atomic="true" data-opened="" data-swipe-direction="right" id="toast-cl-7" class="group block pointer-events-auto relative w-full items-center justify-between space-x-4 overflow-hidden rounded-md p-6 pr-8 shadow-lg transition-all data-[swipe=cancel]:translate-x-0 data-[swipe=end]:translate-x-[var(--kb-toast-swipe-end-x)] data-[swipe=move]:translate-x-[var(--kb-toast-swipe-move-x)] data-[swipe=move]:transition-none data-[opened]:animate-in data-[closed]:animate-out data-[swipe=end]:animate-out data-[closed]:fade-out-80 data-[closed]:slide-out-to-right-full data-[opened]:slide-in-from-top-full data-[opened]:sm:slide-in-from-bottom-full border-l-4 !border-t-gray-200 !border-r-gray-200 !border-b-gray-200 border border-green-700" aria-labelledby="toast-cl-7-title" aria-describedby="toast-cl-7-description"><div id="toast-cl-7-title">Test title</div><div id="toast-cl-7-description" class="text-sm !ml-0">Test description</div></li></div>"`;
